src/api/forms/FormsAPI.js

```
@@ -162,28 +145,91 @@ export default class FormsAPI extends EventEmitter {
                     key = property.join('.');
                 }
 
-                _.set(changes, `${key}`, data.value);
-                self.emit('formPropertiesChanged', changes);
+                _.set(changes, key, data.value);
             }
         }
 
-        //This is not called when a user dismisses a form by pressing escape or the close dialog button
+        //The promise is never resolved/rejected when a user dismisses an overlay by pressing escape or using the [x] button
         function onFormAction(callback) {
             return () => {
-                if (element) {
-                    formElement.remove();
-                } else {
-                    overlay.dismiss();
-                }
+                overlay.dismiss();
 
                 if (callback) {
                     callback(changes);
                 }
             };
         }
 
-        function onFormDismiss(callback) {
+        return promise;
+    }
+
+    /**
+     * Show form as a child of the element provided with given form structure
+     *
+     * @public
+     * @param {Array<Section>} formStructure a form structure, array of section
+     * @param {Object} options
+     *      @property {HTMLElement} element Parent Element to render a Form
+     *      @property {function} onChange a callback function when any changes detected
+     */
+    showCustomForm(formStructure, {
+        element,
+        onChange
+    } = {}) {
+        if (element === undefined) {
```

Let's just throw an error if they don't provide `element`, it's required.